sshd-core/src/main/java/org/apache/sshd/common/config/keys/KeyUtils.java

```diff
@@ -166,6 +167,69 @@ public static PosixFilePermission validateStrictKeyFilePermissions(Path path, Li
     }
 
     /**
+     * <P>Checks if a path has strict ownership</P>
+     * <UL>
+     *
+     * <LI><P>
+     * The path must be owned by current user.
+     * </P></LI>
+     *
+     * <LI><P>
+     * (For {@code Unix}) The path may be owned by root.
+     * </P></LI>
+     *
+     * <LI><P>
+     * (For {@code Unix}) If the path is a file, then its folder must also
+     * have valid owner.
+     * </P></LI>
+     *
+     * </UL>
+     *
+     * @param path    The {@link Path} to be checked - ignored if {@code null}
+     *                or does not exist
+     * @param options The {@link LinkOption}s to use to query the file's permissions
+     * @return The violated owner - {@code null} if
+     * no violations detected
+     * @throws IOException If failed to retrieve the ownership
+     */
+    public static String validateStrictKeyFileOwner(Path path, LinkOption... options) throws IOException {
+        if ((path == null) || (!Files.exists(path, options))) {
+            return null;
+        }
+
+        String current = IoUtils.getCurrentUser();
+        String owner = IoUtils.getFileOwner(path, options);
+
+        if (current == null &&  owner == null) {
+            // we cannot detect permissions
+            return null;
+        }
+
+        Set<String> expected = new HashSet<>();
```

Comment:
did not do this as root is case sensitive and this should not be supported on Windows.